src/appleseed/renderer/kernel/rendering/final/adaptivetilerenderer.cpp

```diff
@@ -298,24 +292,23 @@ namespace
             create_rendering_blocks(rendering_blocks, padded_tile_bbox, framebuffer->get_crop_window());
 
             // First uniform pass based on adaptiveness parameter.
-            if (m_params.m_adaptiveness < 1.0f)
+            if (m_params.m_min_samples > 0)
             {
+                const size_t batch_size = m_params.m_max_samples > 0 ?
+                    min(m_params.m_min_samples, m_params.m_max_samples) : m_params.m_min_samples;
+
                 for (size_t i = 0; i < rendering_blocks.size(); ++i)
```

Comment:
That loop looks suspicious. I think it works as expected but it would be a lot clearer if you replaced it by something like

    while (!rendering_blocks.empty())
    {
        ...
    }